{"path":"lucene/facet/src/test/org/apache/lucene/util/collections/IntArrayTest#testSet().mjava","commits":[{"id":"b89678825b68eccaf09e6ab71675fc0b0af1e099","date":1334669779,"type":1,"author":"Robert Muir","isMerge":false,"pathNew":"lucene/facet/src/test/org/apache/lucene/util/collections/IntArrayTest#testSet().mjava","pathOld":"modules/facet/src/test/org/apache/lucene/util/collections/IntArrayTest#testSet().mjava","sourceNew":"  @Test\n  public void testSet() {\n    int[] original = new int[] { 2,4,6,8,10,12,14 };\n    int[] toSet = new int[] { 1,3,5,7,9,11};\n    \n    IntArray arr = new IntArray();\n    for (int val : original) {\n      arr.addToArray(val);\n    }\n    \n    for (int i = 0; i < toSet.length; i++ ) {\n      int val = toSet[i];\n      arr.set(i, val);\n    }\n    \n    // Test to see if the set worked correctly\n    for (int i = 0; i < toSet.length; i++ ) {\n      assertEquals(toSet[i], arr.get(i));\n    }\n    \n    // Now attempt to set something outside of the array\n    try {\n      arr.set(100, 99);\n      fail(\"IntArray.set should have thrown an exception for attempting to set outside the array\");\n    } catch (ArrayIndexOutOfBoundsException e) {\n      // We expected this to happen so let it fall through\n      // silently\n    }\n    \n  }\n\n","sourceOld":"  @Test\n  public void testSet() {\n    int[] original = new int[] { 2,4,6,8,10,12,14 };\n    int[] toSet = new int[] { 1,3,5,7,9,11};\n    \n    IntArray arr = new IntArray();\n    for (int val : original) {\n      arr.addToArray(val);\n    }\n    \n    for (int i = 0; i < toSet.length; i++ ) {\n      int val = toSet[i];\n      arr.set(i, val);\n    }\n    \n    // Test to see if the set worked correctly\n    for (int i = 0; i < toSet.length; i++ ) {\n      assertEquals(toSet[i], arr.get(i));\n    }\n    \n    // Now attempt to set something outside of the array\n    try {\n      arr.set(100, 99);\n      fail(\"IntArray.set should have thrown an exception for attempting to set outside the array\");\n    } catch (ArrayIndexOutOfBoundsException e) {\n      // We expected this to happen so let it fall through\n      // silently\n    }\n    \n  }\n\n","bugFix":null,"bugIntro":[],"isBuggy":false,"nexts":[],"revCommit":null},{"id":"607428da722dcb3e86bbd11c63de8986e6275c36","date":1360334150,"type":5,"author":"Shai Erera","isMerge":false,"pathNew":"lucene/facet/src/test/org/apache/lucene/facet/collections/IntArrayTest#testSet().mjava","pathOld":"lucene/facet/src/test/org/apache/lucene/util/collections/IntArrayTest#testSet().mjava","sourceNew":"  @Test\n  public void testSet() {\n    int[] original = new int[] { 2,4,6,8,10,12,14 };\n    int[] toSet = new int[] { 1,3,5,7,9,11};\n    \n    IntArray arr = new IntArray();\n    for (int val : original) {\n      arr.addToArray(val);\n    }\n    \n    for (int i = 0; i < toSet.length; i++ ) {\n      int val = toSet[i];\n      arr.set(i, val);\n    }\n    \n    // Test to see if the set worked correctly\n    for (int i = 0; i < toSet.length; i++ ) {\n      assertEquals(toSet[i], arr.get(i));\n    }\n    \n    // Now attempt to set something outside of the array\n    try {\n      arr.set(100, 99);\n      fail(\"IntArray.set should have thrown an exception for attempting to set outside the array\");\n    } catch (ArrayIndexOutOfBoundsException e) {\n      // We expected this to happen so let it fall through\n      // silently\n    }\n    \n  }\n\n","sourceOld":"  @Test\n  public void testSet() {\n    int[] original = new int[] { 2,4,6,8,10,12,14 };\n    int[] toSet = new int[] { 1,3,5,7,9,11};\n    \n    IntArray arr = new IntArray();\n    for (int val : original) {\n      arr.addToArray(val);\n    }\n    \n    for (int i = 0; i < toSet.length; i++ ) {\n      int val = toSet[i];\n      arr.set(i, val);\n    }\n    \n    // Test to see if the set worked correctly\n    for (int i = 0; i < toSet.length; i++ ) {\n      assertEquals(toSet[i], arr.get(i));\n    }\n    \n    // Now attempt to set something outside of the array\n    try {\n      arr.set(100, 99);\n      fail(\"IntArray.set should have thrown an exception for attempting to set outside the array\");\n    } catch (ArrayIndexOutOfBoundsException e) {\n      // We expected this to happen so let it fall through\n      // silently\n    }\n    \n  }\n\n","bugFix":null,"bugIntro":[],"isBuggy":false,"nexts":[],"revCommit":null}],"commit2Parents":{"607428da722dcb3e86bbd11c63de8986e6275c36":["b89678825b68eccaf09e6ab71675fc0b0af1e099"],"b89678825b68eccaf09e6ab71675fc0b0af1e099":["a0e7ee9d0d12370e8d2b5ae0a23b6e687e018d85"],"a0e7ee9d0d12370e8d2b5ae0a23b6e687e018d85":[],"cd5edd1f2b162a5cfa08efd17851a07373a96817":["607428da722dcb3e86bbd11c63de8986e6275c36"]},"commit2Childs":{"607428da722dcb3e86bbd11c63de8986e6275c36":["cd5edd1f2b162a5cfa08efd17851a07373a96817"],"b89678825b68eccaf09e6ab71675fc0b0af1e099":["607428da722dcb3e86bbd11c63de8986e6275c36"],"a0e7ee9d0d12370e8d2b5ae0a23b6e687e018d85":["b89678825b68eccaf09e6ab71675fc0b0af1e099"],"cd5edd1f2b162a5cfa08efd17851a07373a96817":[]},"heads":["cd5edd1f2b162a5cfa08efd17851a07373a96817"],"roots":["a0e7ee9d0d12370e8d2b5ae0a23b6e687e018d85"],"pathCommit":null}